modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -485,6 +473,31 @@ object SeqexecEngine extends SeqexecConfiguration {
     )
   }
 
+  def nextRunnableObservations(qid: QueueId)(st: EngineState): Set[Observation.Id] = {
+
+    def findRunnables(used: Set[Resource], obs: Map[Observation.Id, Set[Resource]], acc: Set[Observation.Id]): Set[Observation.Id] = {
```

Comment:
Can you add a comment about what this does? Are you sure it can't be simplified?

Reply:
I guess so. One obvious optimization I see is that `findRunnables` only needs to check the obs that have not being checked yet.

Reply:
Looks like it could be done with a fold over the `Map`.